dist/src/main/java/io/camunda/application/commons/console/ping/PingConsoleTask.java

```diff
@@ -42,9 +43,15 @@ public PingConsoleTask(
     this.pingConfiguration = pingConfiguration;
     this.client = client;
     final var retryConfiguration = new RetryConfiguration();
-    retryConfiguration.setMaxRetries(NUMBER_OF_MAX_RETRIES);
-    retryConfiguration.setRetryDelayMultiplier(RETRY_DELAY_MULTIPLIER);
-    retryConfiguration.setMaxRetryDelay(MAX_RETRY_DELAY);
+    retryConfiguration.setMaxRetries(
+        Optional.ofNullable(pingConfiguration.retryConfiguration().numberOfMaxRetries())
+            .orElse(NUMBER_OF_MAX_RETRIES));
+    retryConfiguration.setRetryDelayMultiplier(
+        Optional.ofNullable(pingConfiguration.retryConfiguration().retryDelayMultiplier())
+            .orElse(RETRY_DELAY_MULTIPLIER));
+    retryConfiguration.setMaxRetryDelay(
+        Optional.ofNullable(pingConfiguration.retryConfiguration().maxRetryDelay())
+            .orElse(MAX_RETRY_DELAY));
```

Comment:
❓ Have you checked that this actually works, e.g. by running with partial config? I suspect it doesn't work because the record uses primitive types which can't be null, thus preventing spring from constructing the object.